aws/resource_aws_api_gateway_vpc_link_test.go

```diff
@@ -0,0 +1,139 @@
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckAwsAPIGatewayVpcLinkDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAPIGatewayVpcLinkConfig(rName),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAwsAPIGatewayVpcLinkExists("aws_api_gateway_vpc_link.test"),
+					resource.TestCheckResourceAttr("aws_api_gateway_vpc_link.test", "name", fmt.Sprintf("tf-apigateway-%s", rName)),
+					resource.TestCheckResourceAttr("aws_api_gateway_vpc_link.test", "description", "test"),
+					resource.TestCheckResourceAttr("aws_api_gateway_vpc_link.test", "target_arns.#", "2"),
+				),
+			},
+			{
+				Config: testAccAPIGatewayVpcLinkConfig_Update(rName),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAwsAPIGatewayVpcLinkExists("aws_api_gateway_vpc_link.test"),
+					resource.TestCheckResourceAttr("aws_api_gateway_vpc_link.test", "name", fmt.Sprintf("tf-apigateway-update-%s", rName)),
+					resource.TestCheckResourceAttr("aws_api_gateway_vpc_link.test", "description", "test update"),
+					resource.TestCheckResourceAttr("aws_api_gateway_vpc_link.test", "target_arns.#", "2"),
+				),
+			},
+		},
+	})
+}
+
+func testAccCheckAwsAPIGatewayVpcLinkDestroy(s *terraform.State) error {
+	conn := testAccProvider.Meta().(*AWSClient).apigateway
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "aws_api_gateway_vpc_link" {
+			continue
+		}
+
+		input := &apigateway.GetVpcLinkInput{
+			VpcLinkId: aws.String(rs.Primary.ID),
+		}
+
+		_, err := conn.GetVpcLink(input)
+		if err != nil {
+			if isAWSErr(err, apigateway.ErrCodeNotFoundException, "") {
+				return nil
+			}
+			return err
+		}
+
+		return fmt.Errorf("Expected VPC Link to be destroyed, %s found", rs.Primary.ID)
+	}
+
+	return nil
+}
+
+func testAccCheckAwsAPIGatewayVpcLinkExists(name string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		rs, ok := s.RootModule().Resources[name]
+		if !ok {
+			return fmt.Errorf("Not found: %s", name)
+		}
+
+		conn := testAccProvider.Meta().(*AWSClient).apigateway
+
+		input := &apigateway.GetVpcLinkInput{
+			VpcLinkId: aws.String(rs.Primary.ID),
+		}
+
+		_, err := conn.GetVpcLink(input)
+		if err != nil {
+			return err
+		}
+
+		return nil
+	}
+}
+
+func testAccAPIGatewayVpcLinkConfig_basis(rName string) string {
+	return fmt.Sprintf(`
+resource "aws_lb" "test_a" {
+  name = "tf-lb-%s"
+  internal = true
+  load_balancer_type = "network"
+  subnets = ["${aws_subnet.test.id}"]
+}
+
+resource "aws_lb" "test_b" {
+  name = "tf-lb-%s"
+  internal = true
+  load_balancer_type = "network"
+  subnets = ["${aws_subnet.test.id}"]
+}
+
+resource "aws_vpc" "test" {
+  cidr_block = "10.10.0.0/16"
+}
+
+data "aws_availability_zones" "test" {}
+
+resource "aws_subnet" "test" {
+  vpc_id = "${aws_vpc.test.id}"
+  cidr_block = "10.10.0.0/21"
+  availability_zone = "${data.aws_availability_zones.test.names[0]}"
+}
+`, rName, rName)
```

Comment:
The main reason the test is failing the way it's failing is because two resources (`aws_lb.test_a` and `aws_lb.test_b`) both have the same name and the `TypeSet` will internally merge them into a single ARN since the ARN is the same.

Reply:
Once you fix that you get more reasonable error which I think we can just present to the user:

```
* aws_api_gateway_vpc_link.test: BadRequestException: More than one target arn specified for vpc link.
  status code: 400, request id: d4eabe6c-0ff8-11e8-bea2-d9187e5e8757
```

As you rightly mentioned Amazon currently doesn't support more than 1 LB though, so we'll have to keep this test focused on single LB only.

Reply:
That's right! Thank you for correct.